### PR TITLE
Implement gametest system

### PR DESCRIPTION
**Pull request by marcus8448**
Adds pipe and wire network connection/disconnection tests and a grating flow test

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -55,22 +55,41 @@ plugins {
...
 
 loom {
     accessWidenerPath.set(project.file("src/main/resources/galacticraft.accesswidener"))
     mixin {
-        add("main", "galacticraft.refmap.json")
+        add(sourceSets.main.get(), "galacticraft.refmap.json")
+    }
+
+    runs {
+        register("gametest") {
```
Comment (JoeZwet) [inline build.gradle.kts:85]: can we get a client gametest run too?

**Review by JoeZwet — CHANGES_REQUESTED** [review]
just the one thing

File: build.gradle.kts (hunk 1 above)
Comment (JoeZwet) [reply build.gradle.kts:85]: for local testing in dev envs

Comment (marcus8448) [reply build.gradle.kts:85]: Added in 850f716c0818ed4267c7c1274ae7377874606fc8

**Review by JoeZwet — APPROVED** [review]

**Pull request merged**